Makefile

```diff
@@ -1301,6 +1301,15 @@ $(GINKGO): # Build ginkgo from tools folder.
 $(GOLANGCI_LINT): # Build golangci-lint from tools folder.
 	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) $(GOLANGCI_LINT_PKG) $(GOLANGCI_LINT_BIN) $(GOLANGCI_LINT_VER)
 
+.PHONY: provider-issues
+provider-issues: # Creates issues for all CAPI providers 
+	@if [ -n "${GITHUB_ISSUE_OPENER_TOKEN}" ]; then \
+	go run ./hack/tools/update_providers/provider_issues.go --dryRun=false --token=${GITHUB_ISSUE_OPENER_TOKEN}; \
```

Comment:
Can we default dry-run to false in this target and then set it via env variable. That will help to ensure this doesn't get called accidentally.

Reply:
Not sure I understand correctly. Shall we default the dry-run to `true` you mean?

Reply:
Updated the code to always default to dry-run mode unless an env variable `PROVIDER_ISSUES_DRY_RUN` is set to `"false"`.